ocs_ci/ocs/cluster.py

```diff
@@ -924,3 +939,219 @@ def validate_pg_balancer():
+    reached. Then, the function deletes the pods that are not needed as they
+    are the difference between the limit (100%) and the target percentage
+    (the default target percentage is 30%). This leaves the number of pods needed
+    running IO for cluster throughput to be around the desired percentage.
+
+    Args:
+        pvc_factory (function): A call to pvc_factory function
+        pod_factory (function): A call to pod_factory function
+        target_percentage (float): The percentage of cluster load that is required
+
+    Returns:
+         list: Pod objects that are running IO
+
+    """
+    cl_obj = CephCluster()
+    pvc_objs = list()
+    pod_objs = list()
+    io_run_time = 100**3
+
+    if target_percentage < 0.1:
+        logger.info(
+            f"Target percentage is {target_percentage} which is less "
+            f"than 10%. Hence, IO will not start"
+        )
+        return None
+
+    pvc_size = int(get_osd_pods_memory_sum() * 0.75)
+
+    # Creating 1 pod with a PVC size that equals to the memory
+    # sum of all the OSD pods and starting IO
+    pvc_obj = pvc_factory(
+        interface=constant.CEPHBLOCKPOOL, size=pvc_size,
+        volume_mode=constants.VOLUME_MODE_BLOCK
+    )
+    pvc_objs.append(pvc_obj)
+    pod_obj = pod_factory(
+        pvc=pvc_obj, pod_dict_path=constants.CSI_RBD_RAW_BLOCK_POD_YAML,
+        raw_block_pv=True
+    )
+    pod_objs.append(pod_obj)
+
+    io_file_size = f"{pvc_size-1}G"
+    # 'runtime' is set with a large value of seconds
+    # to make sure that the pods are running
+    pod_obj.run_io(
+        storage_type='block', size=io_file_size,
+        runtime=io_run_time, rate='200M', bs='128K', rw_ratio=25
+    )
+
+    cluster_limit = None
+    time_to_wait = 60 * 30
+    time_before = time.time()
+    logger.info("Determining the cluster throughput limit")
+
+    while not cluster_limit:
+
+        pvc_size = int(get_osd_pods_memory_sum() * 0.25)
+        pvc_obj = pvc_factory(
+            interface=constant.CEPHBLOCKPOOL, size=pvc_size,
+            volume_mode=constants.VOLUME_MODE_BLOCK
+        )
+        pvc_objs.append(pvc_obj)
+        pod_obj = pod_factory(
+            pvc=pvc_obj, pod_dict_path=constants.CSI_RBD_RAW_BLOCK_POD_YAML,
+            raw_block_pv=True
+        )
+        pod_objs.append(pod_obj)
+
+        throughput_before_new_pod = cl_obj.calc_average_throughput()
+        logger.info(
+            f"The cluster throughput BEFORE starting IOs on the newly created pod is "
+            f"{throughput_before_new_pod}"
+        )
+
+        # 'runtime' is set with a large value of seconds to
+        # make sure that the pods are running
+        io_file_size = f"{pvc_size - 1}G"
+        pod_obj.run_io(
+            storage_type='block', size=io_file_size,
+            runtime=io_run_time, rate='200M', bs='128K', rw_ratio=25
+        )
+
+        logger.info(f"Waiting for IO to kick in on pod {pod_obj.name}")
+        time.sleep(20)
```

Comment:
pre-fill of `fio` may take more than 20 seconds for a large pvc size. During this time, it uses most of the bandwidth and is not restricted by the 'rate' option. 
Maybe, we need a function to check if pre-fill part of fio is completed.

Reply:
yes, we can create such a function, maybe in a follow up PR. I think it is less urgent as the PVC are being create with a size of ~6GB, see line 1044:
`pvc_size = int(get_osd_pods_memory_sum() * 0.25)`